### ISSUE TITLE
Subclasses with class properties (without constructor) error

### ISSUE THREAD

**Issue by thejameskyle** [babel/babel#2942]
**Input:**

```js
class MyClass extends ParentClass {
  property = foo;
}
```

**Error:**
```js
SyntaxError: index.js: 'super.*' is not allowed before super() (This is an error on an internal node. Probably an internal error)
undefined
```

**Notes:**
- Doesn't happen with `constructor() { super() }`
- Needs `ParentClass` to occur
- Doesn't matter what the value of the class property is.

Example repo: https://github.com/thejameskyle/__babel-bug-2942

**Comment by thejameskyle** [comment]
Locking this issue because there's been way too much irrelevant discussion around this because people are confusing multiple issues. I'm closing them and opening short detailed issues.

**Comment by sebmck** [comment]
Fixed as of 6.1.18.
